src/target.c

```diff
@@ -53,6 +54,9 @@
 /* global static variables */
 static config_atomic_ptr_t current_target = config_atomic_ptr_initializer;
 static config_atomic_ptr_t default_target = config_atomic_ptr_initializer;
+static config_atomic_ptr_t cons_stream = config_atomic_ptr_initializer;
+static config_atomic_bool_t cons_stream_free = config_atomic_bool_true;
```

Comment:
A different option here would be to use an atomic ptr to a `stream_target` struct, which would handle multithreaded access for you once created. However, it would need to be initialized when accessed for the first time, which might be more complicated than this approach. Ultimately it's your decision on which way to go, I just want to provide the option to you for consideration.

Reply:
I guess creating a structure for the same will create unnecessary complications, defining the variables as it is might be a simple way to proceed